packages/eslint-plugin/tests/rules/naming-convention/naming-convention.test.ts

```diff
@@ -768,6 +803,105 @@ ruleTester.run('naming-convention', rule, {
         },
       ],
     },
+    {
+      code: `
+        const obj = {
+          Bar() {
+            return 42;
+          },
+          async async_bar() {
+            return 42;
+          },
+        };
+        class foo {
+          public Bar() {
+            return 42;
+          }
+          public async async_bar() {
+            return 42;
+          }
+        }
+        abstract class foo {
+          public abstract Bar() {
+            return 42;
+          }
+          public abstract async async_bar() {
+            return 42;
+          }
+        }
+      `,
+      parserOptions,
+      options: [
+        {
+          selector: 'memberLike',
+          format: ['camelCase'],
+        },
+        {
+          selector: ['method', 'objectLiteralMethod'],
+          format: ['snake_case'],
+          modifiers: ['async'],
+        },
+        {
+          selector: 'method',
+          format: ['PascalCase'],
+        },
+      ],
+    },
```

Comment:
We generally try to split up test cases so that each test case exercises one behavior. For `naming-conventions`, that's one kind of named thing (property, etc.) per test case.

Test cases generally only give one failure at a time so it can be annoying to keep fixing them as code changes & you re-run tests. Also, it's hard visually to track between big code snippets & the expected errors list.

Reply:
tests split by type